Source/Fields/LMCCylindricalCavity.cc

```diff
@@ -227,6 +233,11 @@ namespace locust
         double vz = tKassParticleXP[5];
         double term1 = fFieldCore->GetBesselNKPrimeZeros(l,m) / GetDimR();
         double term2 = n * LMCConst::Pi() / GetDimL();
+        if ( fCaterpillarCavity )
+        {
+            // Assume n-channels is the same as the number of etalon sections:
+            term2 *= GetNChannels();
```

Comment:
This implementation should work consistently internal to this class, but may run into other issues when called with the CavitySignalGenerator class. See further comments around (new) line 343.